packages/web-components/src/components/va-search-filter/va-search-filter.tsx

```diff
@@ -0,0 +1,346 @@
+import { Component, Host, h, Prop, Event, EventEmitter, Listen, forceUpdate, State, Watch } from '@stencil/core';
+
+export type FilterFacet = {
+  label: string;
+  category: FilterCategory[];
+  id?: number | string;
+  activeFiltersCount?: number;
+};
+
+export type FilterCategory = {
+  label: string;
+  active?: boolean;
+  id?: number | string;
+};
+
+export type Filter = FilterFacet;
+
+interface FilterChangeParams {
+  facetId: number | string;
+  categoryId: number | string;
+  active: boolean;
+}
+
+/**
+ * @componentName Search Filter
+ * @maturityCategory caution
+ * @maturityLevel candidate
+ */
+@Component({
+  tag: 'va-search-filter',
+  styleUrl: 'va-search-filter.scss',
+  shadow: true,
+})
+export class VaSearchFilter {
+
+  /**
+   * The filter header text.
+   */
+  @Prop() header: string = 'Filters';
+
+  /**
+   * Represents a list of filter facets and their categories.
+   * Use a JSON array of objects with label and id properties.
+   */
+  @Prop({ mutable: true }) filterOptions: Filter[] = [];
+
+  /**
+   * The total number of active filters which is updated when the filter changes.
+   */
+  @State() totalActiveFilters: number = 0;
+
+  /**
+   * Watch for changes to filterOptions and update totalActiveFilters
+   */
+  @Watch('filterOptions')
+  filterOptionsChanged(newValue: Filter[]) {
+    if (newValue && newValue.length > 0) {
+      this.totalActiveFilters = this.getTotalActiveFilters();
+      
+      // Update activeFiltersCount for each facet without triggering the watcher again
+      const updatedOptions = newValue.map((facet) => {
+        return {
+          ...facet,
+          activeFiltersCount: VaSearchFilter.getTotalActiveFiltersByFacet(facet),
+        };
+      });
+      
+      // Only update if there's a difference to avoid infinite recursion
+      const needsUpdate = JSON.stringify(updatedOptions) !== JSON.stringify(this.filterOptions);
+      if (needsUpdate) {
+        this.filterOptions = updatedOptions;
+      }
+    }
+  }
+
+  /**
+   * A custom event emitted when the filter changes. The payload will provide all active categories.
+   */
+  @Event() vaFilterChange : EventEmitter<Filter[]>;
+
+  /**
+   * A custom event emitted when the apply filters button is clicked.
+   */
+  @Event() vaFilterApply : EventEmitter<Filter[]>;
+
+  /**
+   * A custom event emitted when the clear all filters button is clicked.
+   */
+  @Event() vaFilterClearAll : EventEmitter<void>;
+
+  @Listen('resize', { target: 'window' })
+  handleResize() {
+    // Corresponds with the --tablet breakpoint size.
+    // https://design.va.gov/foundation/breakpoints
+    this.isDesktop = window.innerWidth > 640;
```

Comment:
I debated what about which breakpoint it should switch between "desktop" and "mobile web" markup. I was also considering if this is something we want to make flexible like a property that could allow a selection of available breakpoints depending on the situation.